packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -73,8 +98,106 @@ class CupertinoNavigationBar extends StatelessWidget implements PreferredSizeWid
   /// True if the nav bar's background color has no transparency.
   bool get opaque => backgroundColor.alpha == 0xFF;
 
+  /// Use iOS 11 style large title navigation bars.
+  ///
+  /// When true, the navigation bar will split into 2 sections. The static
+  /// top 44px section will be wrapped in a SliverPersistentHeader and a
+  /// second scrollable section behind it will show and replace the `middle`
+  /// text in a larger font when scrolled down.
+  ///
+  /// Navigation bars with large titles must be used in a sliver group such
+  /// as [CustomScrollView].
+  final bool largeTitle;
+
   @override
-  Size get preferredSize => const Size.fromHeight(_kNavBarHeight);
+  Size get preferredSize => const Size.fromHeight(_kNavBarPersistentHeight);
+
+  @override
+  Widget build(BuildContext context) {
+    assert(
+      !largeTitle || middle is Text,
+      "largeTitle mode is only possible when 'middle' is a Text widget",
+    );
+
+    if (!largeTitle) {
+      return _wrapWithBackground(
+        backgroundColor: backgroundColor,
+        child: new _CupertinoPersistentNavigationBar(
+          leading: leading,
+          middle: middle,
+          trailing: trailing,
+          actionsForegroundColor: actionsForegroundColor,
+        ),
+      );
+    } else {
+      return new SliverPersistentHeader(
+        pinned: true, // iOS navigation bars are always pinned.
+        delegate: new _CupertinoLargeTitleNavigationBarSliverDelegate(
+          persistentHeight: _kNavBarPersistentHeight + MediaQuery.of(context).padding.top,
+          leading: leading,
+          middle: middle,
+          trailing: trailing,
+          backgroundColor: backgroundColor,
+          actionsForegroundColor: actionsForegroundColor,
+        ),
+      );
+    }
+  }
+}
+
+/// Returns `child` wrapped with background and a bottom border if background color
+/// is opaque. Otherwise, also blur with [BackdropFilter].
+Widget _wrapWithBackground({Color backgroundColor, Widget child}) {
+  final DecoratedBox childWithBackground = new DecoratedBox(
+    decoration: new BoxDecoration(
+      border: const Border(
+        bottom: const BorderSide(
+          color: _kDefaultNavBarBorderColor,
+          width: 0.0, // One physical pixel.
+          style: BorderStyle.solid,
+        ),
+      ),
+      color: backgroundColor,
+    ),
+    child: child,
+  );
+
+  if (backgroundColor.alpha == 0xFF)
+    return childWithBackground;
+
+  return new ClipRect(
+    child: new BackdropFilter(
+      filter: new ImageFilter.blur(sigmaX: 10.0, sigmaY: 10.0),
+      child: childWithBackground,
+    ),
+  );
+}
+
+/// The top part of the nav bar that's never scrolled away.
```

Comment:
Granted it's private, but perhaps a little more detail. i.e. this area is basically padding when expanded, and serves to display the title text when collapsed.